modules/compute/vm-instance/main.tf

```diff
@@ -261,6 +261,17 @@ resource "google_compute_instance" "compute_vm" {
     }
   }
 
+  dynamic "reservation_affinity" {
+    for_each = var.reservation_name == "" ? [] : [1]
+    content {
+      type = "SPECIFIC_RESERVATION"
+      specific_reservation {
+        key    = "compute.googleapis.com/reservation-name"
+        values = [var.reservation_name]
```

Comment:
Did you test it with both short name and `projects/PROJECT_ID/reservations/RESERVATION_NAME` formats?

AFAIK, the `projects/PROJECT_ID/reservations/RESERVATION_NAME` is BulkInsert-specific format and not a proper selfLink (zone is missing), so it may not work here.

Reply:
Yes It has been tested with both formats and works.